uni/lib/controller/fetchers/restaurant_fetcher.dart

```diff
@@ -0,0 +1,89 @@
+import 'package:http/http.dart';
+import 'package:tuple/tuple.dart';
+import 'package:uni/model/entities/restaurant.dart';
+import 'package:uni/model/entities/session.dart';
+
+import 'package:uni/controller/networking/network_router.dart';
+import 'package:uni/controller/parsers/parser_restaurants.dart';
+
+/// Class for fetching the menu
+class RestaurantFetcher {
+  String spreadSheetUrl = 'https://docs.google.com/spreadsheets/d/'
+      '1TJauM0HwIf2RauQU2GmhdZZ1ZicFLMHuBkxWwVOw3Q4';
+  String jsonEndpoint = '/gviz/tq?tqx=out:json';
+
+  // Range containing the meals table
+  // Format: Date(dd/mm/yyyy), Meal("Almoço", "Jantar), Dish("Sopa", "Carne",
+  //         "Peixe", "Dieta", "Vegetariano", "Salada"), Description(String)
+  String range = "A:D";
+
+  // List the Restaurant sheet names in the Google Sheets Document
+  List<String> sheets = ['Cantina'];
+
+  List<String> sigarraMenuEndpoints = [
+    '${NetworkRouter.getBaseUrl('feup')}CANTINA.EMENTASHOW'
+  ];
+
+  // Generate the Gsheets endpoints list based on a list of sheets
+  List<Tuple2<String, String>> buildGSheetsEndpoints(List<String> sheets) {
+    return sheets
+        .map((sheet) => Tuple2(
+            "$spreadSheetUrl$jsonEndpoint&sheet=${Uri.encodeComponent(sheet)}&range=${Uri.encodeComponent(range)}",
+            sheet))
+        .toList();
+  }
+
+  Future<List<Restaurant>> getRestaurants(Session session) async {
```

Comment:
I think that this method flow makes sense, but it is definitely too large, hence needing to be explained in comments. Can you extract the steps?